gammaTask first prototype

sstiefel19 opened the pull request:
before templating data/mc

iarsene commented on Analysis/Tasks/PWGGA/myGammaTask3.cxx:
```diff
@@ -0,0 +1,281 @@
+        
+        Double_t lAlpha = -1.;
+        Double_t lQt = -1.;
+        computeArmenteros(v0, lAlpha, lQt);
```
The Armenteros-Podolanski quantities are already computed and available in the V0Datas table. Please use those. You can access them via: v0.alpha(), v0.qtarm().

iarsene commented on Analysis/Tasks/PWGGA/myGammaTask3.cxx:
```diff
@@ -0,0 +1,281 @@
+            continue;
+        }
+        
+        Double_t lPsiPair = computePsiPair(v0);
```
I suggest that this quantity is also added to the V0Datas table as a dynamic or expression column. However, at the moment this table is produced by a task in LF. So you may want to discuss in the group and possibly also with David Chinellato if you want to introduce a dependency from PWG-LF. One solution could be to promote the lambdakzerobuilder task from LF to become a global task, for more PWGs to use.

sstiefel19 replied:
okay thanks, I will discuss with them

sstiefel19 replied:
Hi @iarsene,
I have a question regarding your comment: You suggest that the PsiPair quantity is added to the V0Datas table. This is one thing.
Another question is whether we want this task to have a dependency on a task (lambdakzerobuilder) in PWG-LF as it is the case now or if lambdakzerobuilder should become a global task. But this question is irrespective from the first issue with PsiPair, right?

iarsene replied:
Hi, 
Regarding psiPair, I think this is something that is mostly up to you. My comment was in the context in which your task is dependent anyway on the LF task producing the V0Datas. So if you would like to continue in this direction, then it makes sense to move also the calculation of psiPair in the strangeness data model, since many of the quantities you need are already there.
Regarding the lambdakzerobuilder question (yes it is independent on the psiPair), i think this is a strategy decision you would need to take within GA at some point. And if you would like to use the lambdakzerobuilder as a V0Datas producer, then indeed i think this task could be promoted (or a new common one be made). However, this would depend also on David, since this task is part of his strangeness analysis and this would require some changes from his side as well.

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -0,0 +1,203 @@
+        lTrackPos.mcParticle().mother0() == lTrackNeg.mcParticle().mother0()) {
+      auto mother = theMcParticles.iteratorAt(lTrackPos.mcParticle().mother0());
+
+      LOGF(info, "SFS mother: %d", mother.pdgCode());
```
This has to be debug I guess

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -0,0 +1,203 @@
+      }
+    }
+  }
+  cout << "SFS total v0s: " << i << " passed all cuts " << nPassed << " ratio " << nPassed / (Float_t)i << " nConfirmedGammas: " << nConfirmedGammas << " purity: " << nConfirmedGammas / (Float_t)nPassed << endl;
```
please use LOGF

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -0,0 +1,203 @@
+WorkflowSpec defineDataProcessing(ConfigContext const& cfgc)
+{
+  return WorkflowSpec{
+    adaptAnalysisTask<ATask>(cfgc, TaskName{"ga-gammaConversionsMC"})};
```
Please change ATask to a meaningful name, and then you can drop the second argument (TaskName)

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.h:
```diff
@@ -0,0 +1,140 @@
+  Configurable<float> fMinR{"fMinR", 5., "fMinR"};
+  Configurable<float> fMaxR{"fMaxR", 180., "fMaxR"};
+
+  Configurable<float> fPIDnSigmaBelowElectronLine{"fPIDnSigmaBelowElectronLine", -3., "fPIDnSigmaBelowElectronLine"};
```
Use meaningful help strings (third argument) for all configurables

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.h:
```diff
@@ -0,0 +1,140 @@
+
+using tracksAndTPCInfo = soa::Join<aod::Tracks, aod::TracksExtra, aod::pidRespTPCEl, aod::pidRespTPCPi, aod::McTrackLabels>;
+
+using std::cout;
```
use LOGF everywhere

jgrosseo reviewed (changes requested):
Thanks for this first version. I have posted some comments, in addition to the ones of @iarsene 

In addition: it is not needed to have an h and cxx file for the task. Put everything in the cxx file (as for the other analysis tasks).

iarsene commented:
@sstiefel19  Is there any update ? i received the 5 days warning before this PR will be automatically closed.

sstiefel19 commented:
Hi @iarsene, @jgrosseo,
I reacted to the comments you had except to the one, that the PsiPair quantity should be added to the V0data table. This I will do now.

sstiefel19 commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -149,40 +148,40 @@ struct GammaConversionsMC {
       float lPhiV0Rec = static_cast<float>(M_PI) + std::atan2(-lV0.py(), -lV0.px());
```
just saw this quantity is also included in the V0s now. so I will take it from there with the next version

Commit:
add gamma conversion task

Commit:
PWGGA/gammaConversionsMC.cxx
add a cut on found over findable tpc clusters
update two tablenames

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -0,0 +1,324 @@
+    }
+
+    theTPCNClsOverFindable = (float)theTrack.tpcNClsFound() / (float)theTrack.tpcNClsFindable();
+    if (theTPCNClsOverFindable < fMinTPCNClsOverFindable) {
```
There is a dynamic column TPCCrossedRowsOverFindableCls. Here you use found clusters over findable.
Two questions:
- isn't the cut with crossed rows the better quantity?
- if not, would it help to add a dynamic column TPCFoundOverFindableCls? In this way you don't have to pass the value by reference out of the function.

sstiefel19 replied:
q1) I used the found/findable since this is what we are using in gamma conversion code in AliPhysics. But so now I also added TPCCrossedRowsOverFindableCls so we can use that one if it's better
q2) yes, I added it now, definitely the cleaner solution

jgrosseo commented on Analysis/Tasks/PWGGA/gammaConversionsMC.cxx:
```diff
@@ -0,0 +1,324 @@
+    {
+      {"IsPhotonSelected", "IsPhotonSelected", {HistType::kTH1F, {{12, -0.5f, 11.5f}}}},
+
+      {"beforeCuts/hPtRec_before", "hPtRec_before", {HistType::kTH1F, {{100, 0.0f, 25.0f}}}},
```
Why not skip _before here, you have them already in the folder "beforeCuts"?

sstiefel19 replied:
I wanted to keep it in case I quickly plot its still directly visible what it is. But if wanted I can of course remove the _before, let me know

jgrosseo replied:
I meant to do: `beforeCuts/hPtRec", "hPtRec_before`
In this case it should be in the title, but not in the object name (but that one is always in the folder, so it is clear)

jgrosseo reviewed (commented):
See my two minor comments for consideration. I start the CI.

Commit:
PWGGA/gammaConversionsMC

add cut on TPCCrossedRowsOverFindableCls
add TPCFoundOverFindableCls to TracksExtra

Commit:
PWGGA/gammaConversionsMC.cxx

remove _before in histo names

sstiefel19 committed:
Update CMakeLists.txt

remove capital letters in executable name
